cmd/readiness/main.go

```diff
@@ -240,61 +240,11 @@ func isInReadyState(health health.Status) bool {
 			return true
 		}
 
-		if mongoDbServerHasStarted(health) {
-			// There should be only one entry reported for this Pod.
-			return processHealth.IsReadyState()
-		}
+		// Every time the process health is created by the agent,
+		// it checks if the MongoDB process is up and populates this field
+		// (https://github.com/10gen/mms-automation/blob/bb72f74a22d98cfa635c1317e623386b089dc69f/go_planner/src/com.tengen/cm/healthcheck/status.go#L43)
+		// So it's enough to check that this value is not the zero-value for int64
+		return processHealth.LastMongoUpTime != 0 && processHealth.IsReadyState()
```

Comment:
Ok, I get it, we don't have to check that the process is started if the process is already on PRIMARY or SECONDARY, great improvement.

However we take into account where the agent won't write this ReplicationStatus, in which case, we'll have to check if one of the steps is `StartFresh` with a result of `success`.

Reply:
The part when the replication status is not present was not touched by this PR. 
This was previously handled already in the code, and the function explicitly has a comment already:

```
// isInReadyState checks the MongoDB Server state. It returns true if the state
// is PRIMARY or SECONDARY.
// This function will always return true if the agent doesn't publish this state.
```

and it does
```
	if processHealth.ReplicaStatus == nil {
			// We always return true if the Agent does not publish mongodb
			// server state
			return true
		}
```

However, we could use the same logic we use for `lastMongoUpTime`

Reply:
I think you are right, if we check for `lastMongoUpTime` that should be enough.